source/common/upstream/thread_aware_lb_impl.h

```diff
@@ -28,6 +29,43 @@ class ThreadAwareLoadBalancerBase : public LoadBalancerBase, public ThreadAwareL
   };
   using HashingLoadBalancerSharedPtr = std::shared_ptr<HashingLoadBalancer>;
 
+  /**
+   * Class for consistent hashing load balancer (CH-LB) with bounded loads.
+   * It is common to both RingHash and Maglev load balancers, because the logic of selecting the
+   * next host when one is overloaded is independent of the CH-LB type.
+   */
+  class BoundedLoadHashingLoadBalancer : public HashingLoadBalancer {
+  public:
+    BoundedLoadHashingLoadBalancer(HashingLoadBalancerSharedPtr hashing_lb_ptr,
+                                   NormalizedHostWeightVector normalized_host_weights,
+                                   uint32_t hash_balance_factor)
+        : normalized_host_weights_map_(initNormalizedHostWeightMap(normalized_host_weights)),
+          hashing_lb_ptr_(std::move(hashing_lb_ptr)),
+          normalized_host_weights_(std::move(normalized_host_weights)),
+          hash_balance_factor_(hash_balance_factor) {
+      ASSERT(hashing_lb_ptr_ != nullptr);
+      ASSERT(hash_balance_factor > 0);
+    }
+    ~BoundedLoadHashingLoadBalancer() = default;
```

Comment:
This whole destructor declaration can go, there's no need to specify the default here. The only time you need to specify a default dtor is when you need to mark it as virtual, which is why you see this on top level abstract classes throughout the code base.

Reply:
Got it, removed it now.